code/modules/mob/living/carbon/life.dm

```diff
@@ -209,15 +210,62 @@
 			hallucination += 10
 		else if(bz_partialpressure > 0.01)
 			hallucination += 5
+
 	//TRITIUM
 	if(breath_gases[/datum/gas/tritium])
 		var/tritium_partialpressure = (breath_gases[/datum/gas/tritium][MOLES]/breath.total_moles())*breath_pressure
 		radiation += tritium_partialpressure/10
+
 	//NITRYL
-	if (breath_gases[/datum/gas/nitryl])
+	if(breath_gases[/datum/gas/nitryl])
 		var/nitryl_partialpressure = (breath_gases[/datum/gas/nitryl][MOLES]/breath.total_moles())*breath_pressure
 		adjustFireLoss(nitryl_partialpressure/4)
 
+	//MIASMA
+	if(breath_gases[/datum/gas/miasma])
+		var/miasma_partialpressure = (breath_gases[/datum/gas/miasma][MOLES]/breath.total_moles())*breath_pressure
+
+		if(prob(1 * miasma_partialpressure))
+			var/datum/disease/advance/miasma_disease = new/datum/disease/advance
+			miasma_disease.symptoms = miasma_disease.GenerateSymptoms(1, 3)
+			ForceContractDisease(miasma_disease, TRUE, TRUE)
+
+		//Clearing out mood events if insufficient partial pressure
+		if(miasma_partialpressure < 25)
+			END_SIGNAL(src, COMSIG_CLEAR_MOOD_EVENT, "nauseating_stench")
+		if(miasma_partialpressure < 10)
+			END_SIGNAL(src, COMSIG_CLEAR_MOOD_EVENT, "bad smell")
+
+		//Miasma side effects
+		switch(miasma_partialpressure)
+			if(1.5 to 10)
+				// At lower pp, give out a little warning
+				if(prob(5))
+					to_chat(src, "<span class='notice'>There is an unpleasant smell in the air.</span>")
+			if(10 to 25)
+				//At somewhat higher pp, warning becomes more obvious
+				if(prob(15))
+					to_chat(src, "<span class='warning'>You smell something horribly decayed inside this room.</span>")
+					SEND_SIGNAL(src, COMSIG_ADD_MOOD_EVENT, "bad smell", /datum/mood_event/disgust/bad_smell)

```

Comment:
"bad smell" should be used for all the add_mood_events

Reply:
Some smells are worse than others, which makes the mood go lower. It is also seen in how the side effect becomes stronger.

Reply:
Yes, but you should only have one moodlet of a category at a time, the category would be "bad smell".

Right now, if the smell gets worse, youd get two moodlets.